### PR TITLE
update for WinUI .net6

### PR DESCRIPTION
**Pull request by ChrisPulman**
**What kind of change does this PR introduce?**

update

**What is the current behaviour?**

supports .net5

**What is the new behaviour?**

supports .net6

**What might this PR break?**

**Please check if the PR fulfils these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ReactiveUI.WinUI/ActivationForViewFetcher.cs [hunk 1]
```diff
@@ -54,5 +46,9 @@ public IObservable<bool> GetActivationForView(IActivatableView view)
                 .Switch()
                 .DistinctUntilChanged();
         }
+
+        private static void Handler1(object sender, RoutedEventArgs e, Action<bool> eventHandler) => eventHandler(false);
```
Comment (glennawatson) [inline src/ReactiveUI.WinUI/ActivationForViewFetcher.cs]: Better with local function

File: src/ReactiveUI/ReactiveUI.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="MSBuild.Sdk.Extras">
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net5.0;Xamarin.iOS10;Xamarin.Mac20;Xamarin.TVOS10;MonoAndroid11.0;tizen40;net6.0;net6.0-android;net6.0-ios;net6.0-tvos;net6.0-macos;netcoreapp3.1</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);net461;net472;uap10.0.16299</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);net461;net472;uap10.0.16299;net5.0-windows10.0.19041.0;net6.0-windows10.0.19041.0</TargetFrameworks>
```
Comment (glennawatson) [inline src/ReactiveUI/ReactiveUI.csproj:4]: Remove please I just went through the effort of getting rid of windows from the main package

File: src/ReactiveUI/ReactiveUI.csproj [hunk 3]
```diff
@@ -83,7 +83,7 @@
     <Compile Include="Platforms\xamarin-common\**\*.cs" />
   </ItemGroup>
 
-  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) or $(TargetFramework.StartsWith('net5')) or '$(TargetFramework)' == 'net6.0' or $(TargetFramework.StartsWith('netcoreapp3')) or $(TargetFramework.StartsWith('uap')) ">
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) or $(TargetFramework.StartsWith('net5')) or '$(TargetFramework)' == 'net6.0' or $(TargetFramework.StartsWith('net6.0-windows10.0')) or $(TargetFramework.StartsWith('netcoreapp3')) or $(TargetFramework.StartsWith('uap')) ">
```
Comment (glennawatson) [inline src/ReactiveUI/ReactiveUI.csproj:86]: Remove

File: src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj [hunk 4]
```diff
@@ -17,5 +17,4 @@
   <ItemGroup>
     <ProjectReference Include="..\ReactiveUI\ReactiveUI.csproj" />
   </ItemGroup>
-
 </Project>
```
Comment (glennawatson) [inline src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj:20]: Whitespace change only so probably doesnt need to be here

File: src/Directory.build.targets [hunk 5]
```diff
@@ -26,6 +26,12 @@
     <TargetPlatformVersion>10.0.16299.0</TargetPlatformVersion>
     <DefineConstants>$(DefineConstants);NETFX_CORE;XAML;WINDOWS_UWP</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('net5.0-windows10.0'))">
+    <TargetPlatformVersion>10.0.19041.0</TargetPlatformVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('net6.0-windows10.0'))">
```
Comment (glennawatson) [inline src/Directory.build.targets:32]: This belongs with the project that targets windows stuff

File: src/ReactiveUI.WinUI/ActivationForViewFetcher.cs (hunk 1 above)
Comment (ChrisPulman) [reply src/ReactiveUI.WinUI/ActivationForViewFetcher.cs]: Reverted to local function

File: src/ReactiveUI/ReactiveUI.csproj (hunk 2 above)
Comment (ChrisPulman) [reply src/ReactiveUI/ReactiveUI.csproj]: reverted

File: src/ReactiveUI/ReactiveUI.csproj (hunk 3 above)
Comment (ChrisPulman) [reply src/ReactiveUI/ReactiveUI.csproj]: Reverted

File: src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj (hunk 4 above)
Comment (ChrisPulman) [reply src/ReactiveUI.Splat.Tests/ReactiveUI.Splat.Tests.csproj]: reverted

File: src/Directory.build.targets (hunk 5 above)
Comment (ChrisPulman) [reply src/Directory.build.targets]: Removed

**Review by glennawatson — APPROVED** [review]

**Pull request merged**